backend/src/v5/processors/teamspaces/projects/models/commons/tickets.js

```diff
@@ -72,20 +139,38 @@ const storeFiles = (teamspace, project, model, ticket, binaryData) => Promise.al
 	)),
 );
 
+const processExternalData = async (teamspace, project, model, ticketId, { binaries, groups }) => {
+	const stillUsed = Array.from(groups.stillUsed);
+	if (stillUsed.length) {
+		const existingGroups = await getGroupsByIds(teamspace, project, model, ticketId,
+			stillUsed.map(stringToUUID), { _id: 1 });
+
+		if (existingGroups.length !== stillUsed.length) {
```

Comment:
I'd prefer it lives here because we're processing on the data instead of checking the validity of the data.

The distinction is that middleware cleans up the data, processor manipulates the data to a way we want it stored.